pkg/controller/servicebinding/secret.go

```diff
@@ -27,17 +33,90 @@ func (s *secret) buildResourceClient() dynamic.ResourceInterface {
 	return s.client.Resource(gvr).Namespace(s.ns)
 }
 
-// createOrUpdate will take informed payload and either create a new secret or update an existing
-// one. It can return error when Kubernetes client does.
-func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1.OwnerReference) (*unstructured.Unstructured, error) {
+// check if payload secret is same as SBR Secret
+func (s *secret) isSame(SBRSecret string) bool {
 	logger := s.logger.WithValues("Namespace", s.ns, "Name", s.name)
+
+	// fetch SBR secret
+	resourceClient := s.buildResourceClient()
+	existingSecret, err := resourceClient.Get(
+		SBRSecret,
+		metav1.GetOptions{},
+	)
+	if err != nil {
+		logger.Error(err, "Error fetching secret")
+		return false
+	}
+	existingSecretData, _, _ := unstructured.NestedMap(existingSecret.Object, "data")
+
+	// fetch new payload secret
+	payloadInterim := make(map[string]interface{})
+	for k, v := range s.data {
+		payloadInterim[k] = base64.StdEncoding.EncodeToString(v)
+	}
+
+	comparisonResult := nestedMapComparison(existingSecretData, payloadInterim)
+
+	if comparisonResult.Success {
+		logger.Debug("Secret is same.")
+		return true
+	}
+
+	return false
+}
+
+// secretDataHash creates hash of secret data using sha1
+func secretDataHash(data map[string][]byte) (string, error) {
```

Comment:
There's no reason to name this after `secret`, it is only a utility function to get a stable checksum of a  `map[string][]byte`; perhaps the name could be added as well, and it could return the whole string:

```
func HashedResourceNameForData(resourceName string, data map[string][]byte) (string, error) {...}

resourceName, err := HashedResourceNameForData("foo", data)
```

Reply:
Changed it here https://github.com/redhat-developer/service-binding-operator/pull/750/files#diff-0141e5e3383464862fa2299425f3318875159d89221a6cd2f36088f6d06d75b1R37